test/compress/dead-code.js

```diff
@@ -68,10 +68,9 @@ dead_code_constant_boolean_should_warn_more: {
         side_effects : true,
     };
     input: {
-        while (!((foo && bar) || (x + "0"))) {
+        while (!(foo || (x + "0"))) {
             console.log("unreachable");
             var foo;
-            function bar() {}
```

Comment:
This actually flags an issue with the code changes in this PR. Consider the following:
```js
while (console.log(bar)) {
    function bar() {}
}
```
Passing it to Node.js 7 actually prints `undefined` instead of throwing `ReferenceError: bar is not defined`.